src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeKernelMap.cs

```diff
@@ -101,7 +101,7 @@ protected virtual void Dispose(bool disposing)
         /// Returns a <see cref="ResizeKernel"/> for an index value between 0 and DestinationSize - 1.
         /// </summary>
         [MethodImpl(InliningOptions.ShortMethod)]
-        internal ref ResizeKernel GetKernel(int destIdx) => ref this.kernels[destIdx];
+        internal ref ResizeKernel GetKernel(nint destIdx) => ref this.kernels[destIdx];
```

Comment:
For the plain array access `nint` doesn't bring any advantage.
Here the JIT misses a optimization (a bug is tracked, but can't find it at the moment).

If you want best machine code here, look at [Sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgBEBLbAcwDsJcGDmFwBuGvSasASgFdeQ/DBYBhCPgAOHADYwoAZT0A3YTDETGzFnIUclLAJIK9EDYagmwZ8dRodnULzY2kykDCoMAN40DLEMGlAcRtgYMAyw2AAmELzaAJ4M0mYcAF4wANJ6vDDaANoAugwA+gDWVTXm1HEMMXHEAMzpMABmhcVllYE1DADiMBiT1doAggAU/hgMmWYYDpkIAJQMALwAfEOjre3auLXbgnsI9T7dAxdjuKUV17PzizUAIVWvA2Wx2jyOZ3eVymNzu4P2z16sTesFGRU+Ex+cwW1xUwNB912+wOyKiZO6bBgwFk3BYy1wuD0GFWRMeDFOxwYjAAZDywQ99gwADzNNqw3AsAAyMF43AwAAsDi9unE0R8vv8QuKlkVRlz1QBZGD4aB5Q3YKC4BXBFg45ZQKDYPJsFLYPV6WVeVYwpa4ZUUvoAdneAFVeLhsMNlMtMplVuqdTU9WgBcTDirYgBfGg53zUDZVYJDLI5fIMeSR6MMQRQWRgTYYzXXGjRLpxBJJFJpDLZXIFYbaCApABUDBpw2jUAAChgoD4yQB6Zci3CyfD4S15U5LldODhCYJfXAMbAMaoAdwY/kE2F4XgYEFGirSwqZaTAaOOACIm1jYd+i7nLW9YYCwu6LiKi5rhuW47u2sSFoExZ/t8sLrAoNYYJauy8NsCCpoOw4YGOE5TrOUCpqCuhyoqpIIeSDHdIqHCSvo2FQLh+EnFhOFOPhmaqixkpkXoFE8aJM5zoJzEKqx0qyvKCo8TRSmCXmEGruum5QNumk4ieL68Zx154TACAMMM0AMEZRL+CkHA5OkEAXuBDHLpBwrQdpcFkm8oLsXxZkIGSbaqnE3DzOpNCaW+Pm6fB3Qeb8GCGQqaSqYqj7PulDBJtoblJSuXkwTpekMf5mEyrRCqhYGsSRRg0XUFmQA=) method `GetKernelC`.
It's more C# code and I doubt it's needed here. I'd just go with `int` (or `nint` as you have it now) and wait for the JIT fix.

Reply:
No `MemoryMarshal.GetArrayDataReference` for us yet unfortunately..... One day!